core/src/main/java/io/questdb/cairo/ColumnPurgeJob.java

```diff
@@ -231,7 +231,7 @@ private void processTableRecords(CairoEngine engine) {
                     count++;
                     long ts = rec.getTimestamp(0);
                     if (ts != lastTs || taskRun == null) {
-                        if (taskRun != null) {
+                        if (taskRun != null && !taskRun.isEmpty()) {
```

Comment:
isn't this leaking `taskRun` objects? 
say `taskRun` is not null, but it's empty. (=no name set, because token was null before) 
 
this means you had previously `pop()`-ed  from the pool. and now you take another one, without returning the original back. 

am I missing anything?

Reply:
Good spot, fixed